File: src/testFile3.c
#include "thing.h";

void* myFunction3(TestParameter* t) {
    if (t->foo == 0) {
        return 0;
    }

    size_t memory = 0;
    int[4] mems = {1, 3, 7, 19};
    for (int i = 0; i < t->foo; ++i) {
        memory += mems[(i * t->foo) % 4];
    }

    void* mem = malloc(memory);
    recordMemorySample(mem, memory);

    return mem;
}

File: src/testFile3.ts
import {MemoryInfo, recordMemorySample} from "./types";

export function myFunction3(t: MemoryInfo): Uint8Array | null {
    if (t.foo === 0) {
        return null;
    }

    let len = 0;
    const mems = [1, 3, 7, 19];

    for (let i = 0; i < t.foo; ++i) {
        len += mems[(i * t.foo) % 4] * t.size;
    }

    const mem = new Uint8Array(len);
    if (process.env.RECORD_MEMORY === "true") {
        recordMemorySample(mem);
    }

    return mem;
}


